extensions/amp-youtube/0.1/test/test-amp-youtube.js

```diff
@@ -291,6 +291,17 @@ describe('amp-youtube', function() {
     });
   });
 
+  it('should remove iframe after unlayoutCallback', () => {
+    return getYt({'data-videoid': 'mGENRKrdoGY'}).then(yt => {
+      const placeholder = yt.querySelector('[placeholder]');
+      const obj = yt.implementation_;
+      obj.unlayoutCallback();
+      expect(yt.querySelector('iframe')).to.be.null;
+      expect(obj.iframe_).to.be.null;
```

Comment:
let's also assert that `#playerState, #playerReadyPromise, and #playerReadyResolver` that @jridgewell suggested to add are reset here.